### PR TITLE
fix(Templates): Update `aws-lambda-java-log4j2`

### PR DESCRIPTION
**Pull request by MarinaMeza**
Updated aws-lambda-java-log4j2 to 1.4.0 to address log4j vulnerability

[Issue 10370](https://github.com/serverless/serverless/issues/10370)

Addresses: #10370

### CODE REVIEW

**Comment by pgrzesik** [comment]
Thanks a lot @MarinaMeza 🙇 I've run a quick grep over the repo and it seems like there are some additional templates that use this package that could also use an upgrade:
- `aws-scala-sbt`
- `aws-kotlin-jvm-gradle-kts`
- `aws-kotlin-jvm-maven`
- `aws-java-maven`
- `aws-java-gradle`

Would you be interested in upgrading them too? It would be much appreciated to cover all of them as a single commit with an upgrade.

**Comment by MarinaMeza** [comment]
@pgrzesik Yes, no problem! I'll upgrade those too

**Comment by anwari666** [comment]
Hi @MarinaMeza, thanks for updating those. 

I am having the same issue at my company and saw that this file https://github.com/serverless/serverless/blob/7bf8f1b723cc4f6624df7be3d27f246b3a826140/lib/plugins/aws/invokeLocal/runtimeWrappers/java/pom.xml#L25 has indirect reference to log4j. 

I'm not sure how it will impact the repo, but if it's a simple update then perhaps it can be updated within this commit as well?
Thanks!

File: lib/plugins/create/templates/aws-java-maven/pom.xml [hunk 1]
```diff
@@ -17,7 +17,7 @@
     <dependency>
       <groupId>com.amazonaws</groupId>
       <artifactId>aws-lambda-java-log4j2</artifactId>
-      <version>1.1.0</version>
```
Comment (pgrzesik) [inline lib/plugins/create/templates/aws-java-maven/pom.xml:20]: We should update to `1.5.0`

File: lib/plugins/create/templates/aws-kotlin-jvm-maven/pom.xml [hunk 2]
```diff
@@ -41,7 +41,7 @@
         <dependency>
             <groupId>com.amazonaws</groupId>
             <artifactId>aws-lambda-java-log4j2</artifactId>
-            <version>1.1.0</version>
+            <version>1.4.0</version>
```
Comment (pgrzesik) [inline lib/plugins/create/templates/aws-kotlin-jvm-maven/pom.xml:44]: same here

**Comment by MarinaMeza** [comment]
I've updated all references to aws-lambda-java-log4j2. 
I left aws-lambda-java-log4j as it was since i could not find a newer version

**Review by pgrzesik — CHANGES_REQUESTED** [review]
Thank you @MarinaMeza, it looks great, but there seem to be a conflict - could you rebase your branch on top of current `master` branch?

**Review by pgrzesik — APPROVED** [review]
Thank you @MarinaMeza 🙇

**Pull request merged**